### PR TITLE
fix: correctly report the coverage of changed files when using the workingDirectory

### PR DESCRIPTION
**Pull request by paulswartz**
compare:
- (bad) https://github.com/mbta/rtr/pull/979#issuecomment-1064454846
- (good) https://github.com/paulswartz/data_platform/pull/1#issuecomment-1064571961

Note that the (good) comment includes the coverage of the files.

### CODE REVIEW

File: dist/main/index.js [hunk 1]
```diff
@@ -14536,8 +14537,15 @@ function ownerRepo(url) {
...
+    return path;
+  }
+
+  return path + "/";
+}
+
+async function genhtml(coverageFiles, tmpPath, workingDirectory) {
   const artifactName = core.getInput('artifact-name').trim();
   const artifactPath = path.resolve(tmpPath, 'html').trim();
   const args = [...coverageFiles];
```
Comment (ryan-mahoney) [inline dist/main/index.js]: minor js style, `concat` allows this to be one line, with no mutations:

```
const args = [...coverageFiles].concat(['--output-directory', artifactPath])
```

File: dist/main/index.js [hunk 2]
```diff
@@ -14536,8 +14537,15 @@ function ownerRepo(url) {
...
-  const workingDirectory = core.getInput('working-directory').trim() || './';
+function ensureTrailingSlash(path) {
+  if (path.endsWith("/")) {
+    return path;
+  }
+
+  return path + "/";
+}
+
+async function genhtml(coverageFiles, tmpPath, workingDirectory) {
```
Comment (ryan-mahoney) [inline dist/main/index.js:14481]: minor: some linters would ding that this function was used before it was defined, ie, out of order

**Review by ryan-mahoney — APPROVED** [review]
Looks great, left some minor comments related to style

File: dist/main/index.js (hunk 1 above)
Comment (paulswartz) [reply dist/main/index.js]: Turns you you can also do `[...coverageFiles, '--output-directory', artifactPath]`!

Comment (ryan-mahoney) [reply dist/main/index.js]: of course -- even better!

Comment (ryan-mahoney) [reply dist/main/index.js]: You are probably aware, but JS has this syntax for objects which is similar to Elixir, but I haven't noticed in our code too often in the past:

```
const someObj  = {...anotherObj, abc: "def", [someVar]: "qrs"};
```

**Pull request merged**